state/state.go

```diff
@@ -1729,7 +1729,16 @@ func (st *State) AddRelation(eps ...Endpoint) (r *Relation, err error) {
 		return ops, nil
 	}
 	if err = st.run(buildTxn); err == nil {
-		return &Relation{st, *doc}, nil
+		return &Relation{st, *doc}, err
+	} else if errors.IsAlreadyExists(err) {
+		// If the relation already exists, we'll return it
```

Comment:
This doesn't save the caller anything really. We're still doing the KeyRelation call that they would do, so I'm -1 on this. Please keep with the standard approach of returning AlreadyExists, and have the caller react to that how it will.